core/providers/animefreak/__init__.py

```diff
@@ -0,0 +1,40 @@
+import re
+
+import lxml.html as htmlparser
+
+EPISODE_RE = re.compile(r"(?:https?://)?(?:\S+\.)?animefreak\.tv/watch/([^?&/]+)/episode/episode-\d+")
+STREAM_URL_RE = re.compile(r"(?:https?://)?st\d+\.anime1\.com/(?P<file>.*)")
+
+BASE_URL = "https://www.animefreak.tv/watch/{}"
+BASE_STREAM_URL = "https://st%d.anime1.com/%s"
+
+def send_valid(session, stream_url):
+
+    with session.get(stream_url, stream=True) as response:
+        if response.ok:
+            return stream_url
+    
+    file = STREAM_URL_RE.search(stream_url).group('file')
+    
+    for trial in range(6, 12):
+        with session.get(url := BASE_STREAM_URL % (trial, file), stream=True) as response:
```

Comment:
While the walrus operator is very nice to use, I think effort should be made to remove these as it's a 3.8+ feature. Many LTS distros are still on ~3.6.

Reply:
Those operators are scattered all over the code. Oh boy, that's going to be a pain to remove. I'm sorry if they caused you trouble (you know, upgrading and stuff). I'll replace them as soon as I get enough time.